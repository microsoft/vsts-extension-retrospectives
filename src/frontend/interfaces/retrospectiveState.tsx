import { WorkItemTypeModel } from 'azure-devops-extension-api/WorkItemTrackingProcessDefinitions';

export enum ExceptionCode {
    Unexpected = 0,
    NotInheritedProcess = 1,
}

export interface InitialRetrospectiveState {
  displayBoard: boolean;
  exceptionCode?: ExceptionCode;
  isInheritedProcess: boolean;
  retrospectiveWorkItemType?: WorkItemTypeModel;
}
