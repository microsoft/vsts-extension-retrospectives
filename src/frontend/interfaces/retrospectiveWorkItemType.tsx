import { FieldModel, FieldType, PickListModel, WorkItemTypeClass, WorkItemTypeModel } from 'azure-devops-extension-api/WorkItemTrackingProcessDefinitions';

export const feedbackType: FieldModel = {
    description: 'The type of feedback.',
    id: null,
    name: 'Feedback Type',
    pickList: null,
    type: 1,
    url: null,
};

export const upvotes: FieldModel = {
    description: 'Upvote count',
    id: null,
    name: 'Upvotes',
    pickList: null,
    type: FieldType.Integer,
    url: null,
};

export const retrospectiveWorkItemTypeModel: WorkItemTypeModel = {
    behaviors: [],
    class: WorkItemTypeClass.Custom,
    color: '60af49',
    description: 'Tracks retrospective feedback.',
    icon: 'icon_chat_bubble',
    id: null,
    inherits: null,
    isDisabled: false,
    layout: null,
    name: 'Retrospective',
    states: [],
    url: null,
};

export const feedbackPickList: PickListModel = {
    id: null,
    isSuggested: false,
    items: [{ value: 'Negative', id: null }, { value: 'Neutral', id: null }, { value: 'Positive', id: null }],
    name: 'picklist_3afce441-2c57-4572-860a-8cf9b942c022',
    type: 'String',
    url: null,
};

export enum ExceptionCode {
    Unexpected = 0,
    NotInheritedProcess = 1,
}

export interface InitialRetrospectiveState {
    retrospectiveWorkItemType?: WorkItemTypeModel;
    isInheritedProcess: boolean;
    displayBoard: boolean;
    exceptionCode?: ExceptionCode;
}
