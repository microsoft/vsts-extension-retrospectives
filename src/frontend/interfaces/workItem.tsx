import { WorkItem } from 'azure-devops-extension-api/WorkItemTracking/WorkItemTracking';

export interface WorkItemGroup {
  name: string;
  feedbackType: FeedbackType;
  iconName: string;
  workItems: WorkItemExtended[];
}

export interface WorkItemExtended {
  iconName: string;
  workItem: WorkItem;
  upvotes: number;
  isLinkedForGroup: boolean;
  isParentForGroup: boolean;
}

export enum FeedbackType {
  Neutral = 'Neutral',
  Positive = 'Positive',
  Negative = 'Negative',
}

export enum RelationshipType {
  ReferencedByForward = 'Microsoft.VSTS.TestCase.SharedParameterReferencedBy-Forward',
  ReferencedByReverse = 'Microsoft.VSTS.TestCase.SharedParameterReferencedBy-Reverse',
  Related = 'System.LinkTypes.Related',
}

export type WorkflowPhase = 'Collect' | 'Group' | 'Vote' | 'Discuss' | 'Act';
export const WorkflowPhase = {
    get Collect(): WorkflowPhase { return 'Collect'; },
    get Group(): WorkflowPhase { return 'Group'; },
    get Vote(): WorkflowPhase { return 'Vote'; },
    get Discuss(): WorkflowPhase { return 'Discuss'; },
    get Act(): WorkflowPhase { return 'Act'; },
    getState(str: string): WorkflowPhase {
      if ( str === 'Collect') return WorkflowPhase.Collect;
      if ( str === 'Group') return WorkflowPhase.Group;
      if ( str === 'Vote') return WorkflowPhase.Vote;
      if ( str === 'Discuss') return WorkflowPhase.Discuss;
      if ( str === 'Act') return WorkflowPhase.Act;
      return undefined;
    }
  }

export interface IRetrospectiveItemCreate {
  title: string;
  feedbackType: FeedbackType;
  iteration: string;
  areaPath: string;
  isAnonymous: boolean;
}

export interface IRetrospectiveItemsQuery {
  feedbackType: FeedbackType;
  iteration: string;
  areaPath: string;
}
