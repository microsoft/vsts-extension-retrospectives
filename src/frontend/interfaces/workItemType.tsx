import {WorkItemTypeModel, FieldModel} from 'azure-devops-extension-api/WorkItemTrackingProcessDefinitions';

export interface CustomWorkItemType {
    workItemTypeModel: WorkItemTypeModel;
    customFields: FieldModel[];
}