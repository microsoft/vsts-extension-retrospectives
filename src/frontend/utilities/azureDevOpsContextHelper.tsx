import { getHost } from 'azure-devops-extension-sdk';
import { getHostAuthority } from '../utilities/servicesHelper';

const internalOrgNames = [
  'reflect-retrospective-hackathon',
  'reflect-demo',
  'microsoft',
  'microsoftit',
  'mseng',
  'msazure',
  'onebranch',
];

/**
 * Returns whether the current org in VSTS context is a recognized internal org.
 */
export const isInternalOrg = () => {
  const host = getHost();
  return internalOrgNames.indexOf(host.name.toLowerCase().trim()) !== -1;
};

/**
 * Returns whether the extension is run in a hosted environment (as opposed to an on-premise environment).
 * In Azure DevOps terms, hosted environment is also known as "Azure DevOps Services" and on-premise environment is known as
 * "Team Foundation Server" or "Azure DevOps Server".
 */
export const isHostedAzureDevOps = async () => {
  const hostAuthority = await getHostAuthority();
  return hostAuthority === 'dev.azure.com' || hostAuthority.endsWith('.visualstudio.com');
};
