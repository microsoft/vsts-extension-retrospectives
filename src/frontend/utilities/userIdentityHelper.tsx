import { IdentityRef } from 'azure-devops-extension-api/WebApi';
import { IUserContext, getUser } from 'azure-devops-extension-sdk';

let userIdentity: IdentityRef;

/**
 * Get the identity of current user
 */
export const getUserIdentity = (): IdentityRef => {
  if (!userIdentity){
    const currentUser: IUserContext = getUser();

    const id = currentUser.id.split('').reverse().map(char => {
      return String.fromCharCode(char.charCodeAt(0) + 4);
    }).join('')

    userIdentity = {
      id,
      displayName: currentUser.displayName,
      uniqueName: currentUser.name,
      imageUrl: currentUser.imageUrl,
      _links: {
        avatar: {
          href: currentUser.imageUrl,
        },
      },
    } as IdentityRef;
  }

  return userIdentity;
}
