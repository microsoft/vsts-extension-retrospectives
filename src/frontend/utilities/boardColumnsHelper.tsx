import { IFeedbackColumn } from '../interfaces/feedback';
import { generateUUID } from './random';

export const getColumnsByTemplateId = (templateId: string): IFeedbackColumn[] => {
  switch (templateId) {
    case 'speedboat': // Speedboat retrospective - Propellors, Life Preserver, Anchors, Rocks
    return [
      {
        accentColor: '#008000',
        iconClass: 'fas fa-rocket',
        id: generateUUID(),
        title: 'Propellors'
      },
      {
        accentColor: '#f6af03',
        iconClass: 'fas fa-life-ring',
        id: generateUUID(),
        title: 'Life preserver',
      },
      {
        accentColor: '#F78A53',
        iconClass: 'fas fa-anchor',
        id: generateUUID(),
        title: 'Anchors',
      },
      {
        accentColor: '#cc293d',
        iconClass: 'far fa-exclamation',
        id: generateUUID(),
        title: 'Rocks',
      },
    ];
    case '4ls': // The 4 Ls - Like, Learned, Lacked, Longed For
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Liked'
        },
        {
          accentColor: '#f6af08',
          iconClass: 'fas fa-book',
          id: generateUUID(),
          title: 'Learned',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'far fa-compass',
          id: generateUUID(),
          title: 'Lacked',
        },
        {
          accentColor: '#8063bf',
          iconClass: 'far fa-eye',
          id: generateUUID(),
          title: 'Longed for',
        },
      ];
    case '1to1': // 1-to-1 - Good, So-so, Improve, Done
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Good',
        },
        {
          accentColor: '#f6af08',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'So-so',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-balance-scale-right',
          id: generateUUID(),
          title: 'Improve',
        },
        {
          accentColor: '#8063bf',
          iconClass: 'fas fa-birthday-cake',
          id: generateUUID(),
          title: 'Done',
        },
      ];
    case 'daki': // Drop, Add, Keep, Improve
      return [
        {
          accentColor: '#cc293d',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'Drop',
        },
        {
          accentColor: '#008000',
          iconClass: 'fas fa-smile',
          id: generateUUID(),
          title: 'Add',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-book',
          id: generateUUID(),
          title: 'Keep',
        },
        {
          accentColor: '#8063bf',
          iconClass: 'far fa-compass',
          id: generateUUID(),
          title: 'Improve',
        },
      ];
    case 'mad-sad-glad':
      return [
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-angry',
          id: generateUUID(),
          title: 'Mad',
        },
        {
          accentColor: '#f6af08',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'Sad',
        },
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Glad',
        },
      ];
    case 'good-bad-ideas':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Good',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-angry',
          id: generateUUID(),
          title: 'Bad',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'Ideas',
        },
      ];
    case 'kalm': // Keep, Add, Less, More
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Keep',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-book',
          id: generateUUID(),
          title: 'Add',
        },
        {
          accentColor: '#f6af08',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'Less',
        },
        {
          accentColor: '#8063bf',
          iconClass: 'far fa-comments',
          id: generateUUID(),
          title: 'More',
        },
      ];
    case 'start-stop-continue':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Start',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'Stop',
        },
        {
          accentColor: '#f6af08',
          iconClass: 'far fa-eye',
          id: generateUUID(),
          title: 'Continue',
        },
      ];
    case 'psy-safety':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What makes it safe',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What hinders safety',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'clarity':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What provides clarity',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What hinders clarity',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'energy':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What provides energy',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What drains energy',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'wlb':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What helps work-life balance',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What hinders work-life balance',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'team-confidence':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What increases confidence in team',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What decreases confidence in team',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'team-efficiency':
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'What helps efficiency',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'far fa-frown',
          id: generateUUID(),
          title: 'What hinders efficiency',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'fas fa-exclamation',
          id: generateUUID(),
          title: 'One action in next sprint',
        },
      ];
    case 'wlai': // Went Well, Learned, Impediments, Accelerators
      return [
        {
          accentColor: '#008000',
          iconClass: 'far fa-smile',
          id: generateUUID(),
          title: 'Went Well',
        },
        {
          accentColor: '#8063bf',
          iconClass: 'fas fa-book',
          id: generateUUID(),
          title: 'Learned',
        },
        {
          accentColor: '#0078d4',
          iconClass: 'far fa-compass',
          id: generateUUID(),
          title: 'Accelerators',
        },
        {
          accentColor: '#cc293d',
          iconClass: 'fas fa-question',
          id: generateUUID(),
          title: 'Impediments',
        },
      ];
    default: {
      return [{
        accentColor: '#008000',
        iconClass: 'far fa-smile',
        id: generateUUID(),
        title: 'What went well',
      }, {
        accentColor: '#cc293d',
        iconClass: 'far fa-frown',
        id: generateUUID(),
        title: "What didn't go well",
      }]
    }
  }
};
