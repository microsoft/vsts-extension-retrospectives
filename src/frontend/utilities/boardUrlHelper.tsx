import { getHostBaseUrl, getProjectName } from '../utilities/servicesHelper';

/**
 * Generates a URL-safe deep link for board.
 *
 * @param teamId Id of selected team
 * @param boardId Id of selected board
 *
 * @returns the URL-safe (encoded) URL
 */
export const getBoardUrl = async (teamId: string, boardId: string): Promise<string> => {
  const hostBase = await getHostBaseUrl();
  const projectName = await getProjectName();

  const boardDeepLinkUrl = `${hostBase}${projectName}/_apps/hub/ms-devlabs.team-retrospectives.home#teamId=${teamId}&boardId=${boardId}`;

  return encodeURI(boardDeepLinkUrl);
}
