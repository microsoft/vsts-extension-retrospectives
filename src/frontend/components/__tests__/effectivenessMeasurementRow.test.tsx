import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import EffectivenessMeasurementRow, { EffectivenessMeasurementRowProps } from "../effectivenessMeasurementRow";
import { ITeamEffectivenessMeasurementVoteCollection } from "../../interfaces/feedback";
import { encrypt, getUserIdentity } from "../../utilities/userIdentityHelper";

jest.mock("../../utilities/userIdentityHelper", () => ({
  getUserIdentity: jest.fn(),
  encrypt: jest.fn(),
}));

const mockGetUserIdentity = getUserIdentity as jest.MockedFunction<typeof getUserIdentity>;
const mockEncrypt = encrypt as jest.MockedFunction<typeof encrypt>;

const defaultProps: EffectivenessMeasurementRowProps = {
  questionId: 1,
  title: "Test Question",
  subtitle: "Test subtitle for the question",
  iconClass: "fa-solid fa-magnifying-glass",
  tooltip: "This is a test tooltip",
  votes: [],
  onSelectedChange: jest.fn(),
};

const mockVotes: ITeamEffectivenessMeasurementVoteCollection[] = [
  {
    userId: "encrypted-test-user-id",
    responses: [
      { questionId: 1, selection: 5 },
      { questionId: 2, selection: 8 },
    ],
  },
];

describe("EffectivenessMeasurementRow", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockGetUserIdentity.mockReturnValue({ id: "test-user-id" } as any);
    mockEncrypt.mockReturnValue("encrypted-test-user-id");
  });

  describe("Component Rendering", () => {
    it("renders correctly with default props", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);
      expect(container.firstChild).toBeInTheDocument();
    });

    it("renders correctly with votes data", () => {
      const props = {
        ...defaultProps,
        votes: mockVotes,
      };
      const { container } = render(<EffectivenessMeasurementRow {...props} />);
      expect(container.firstChild).toBeInTheDocument();
    });

    it("renders the correct title and subtitle", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      const titleElement = container.querySelector(".effectiveness-measurement-question-cell span");
      expect(titleElement).toHaveTextContent("Test Question");

      const subtitleElement = container.querySelector(".effectiveness-measurement-question-cell");
      expect(subtitleElement).toHaveTextContent("Test subtitle for the question");
    });

    it("renders the correct icon class", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      const iconElement = container.querySelector(".fa-solid.fa-magnifying-glass");
      expect(iconElement).toBeInTheDocument();
    });

    it("renders tooltip with correct content", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      const tooltipHost = container.querySelector('[data-is-focusable="true"]');
      expect(tooltipHost).toBeInTheDocument();
    });

    it("renders all 10 voting buttons", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      const votingButtons = container.querySelectorAll(".effectivemess-measurement-voting-button");
      expect(votingButtons).toHaveLength(10);
    });

    it("renders voting buttons with correct aria-labels", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      for (let i = 1; i <= 10; i++) {
        const button = container.querySelector(`button[aria-label="${i}"]`);
        expect(button).toBeInTheDocument();
      }
    });
  });

  describe("Accessibility", () => {
    it("provides proper aria-labels for screen readers", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      for (let i = 1; i <= 10; i++) {
        const button = container.querySelector(`button[aria-label="${i}"]`);
        expect(button).toBeInTheDocument();
      }
    });

    it("renders buttons with appropriate accessibility attributes", () => {
      const { container } = render(<EffectivenessMeasurementRow {...defaultProps} />);

      const buttons = container.querySelectorAll("button[aria-label]");
      expect(buttons).toHaveLength(10);

      buttons.forEach((button: any) => {
        expect(button).toHaveAttribute("aria-label");
      });
    });
  });
});
