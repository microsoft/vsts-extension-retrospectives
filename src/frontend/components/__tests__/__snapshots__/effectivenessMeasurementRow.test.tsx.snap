// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffectivenessMeasurementRow Component Rendering renders correctly with default props 1`] = `
<tr
  className="effectiveness-measurement-row"
>
  <td
    className="effectiveness-measurement-question-cell"
  >
    <i
      className="fa-solid fa-magnifying-glass"
    />
      
    <span
      style={
        {
          "fontWeight": "bolder",
        }
      }
    >
      Test Question
    </span>
    <br />
    Test subtitle for the question
  </td>
  <td
    className="effectiveness-measurement-tooltip-cell"
  >
    <StyledTooltipHostBase
      calloutProps={
        {
          "gapSpace": 0,
        }
      }
      content={
        <div
          dangerouslySetInnerHTML={
            {
              "__html": "This is a test tooltip",
            }
          }
        />
      }
      hostClassName="toggle-carousel-button-tooltip-wrapper"
    >
      <CustomizedDefaultButton
        className="contextual-menu-button hide-mobile"
        iconProps={
          {
            "iconName": "Error",
          }
        }
      />
    </StyledTooltipHostBase>
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="1"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="2"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="3"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="4"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="5"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="6"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="7"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="8"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="9"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="10"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
</tr>
`;

exports[`EffectivenessMeasurementRow Component Rendering renders correctly with votes data 1`] = `
<tr
  className="effectiveness-measurement-row"
>
  <td
    className="effectiveness-measurement-question-cell"
  >
    <i
      className="fa-solid fa-magnifying-glass"
    />
      
    <span
      style={
        {
          "fontWeight": "bolder",
        }
      }
    >
      Test Question
    </span>
    <br />
    Test subtitle for the question
  </td>
  <td
    className="effectiveness-measurement-tooltip-cell"
  >
    <StyledTooltipHostBase
      calloutProps={
        {
          "gapSpace": 0,
        }
      }
      content={
        <div
          dangerouslySetInnerHTML={
            {
              "__html": "This is a test tooltip",
            }
          }
        />
      }
      hostClassName="toggle-carousel-button-tooltip-wrapper"
    >
      <CustomizedDefaultButton
        className="contextual-menu-button hide-mobile"
        iconProps={
          {
            "iconName": "Error",
          }
        }
      />
    </StyledTooltipHostBase>
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="1"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="2"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="3"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="4"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="5"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleFill",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="6"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="7"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="8"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="9"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
  <td>
    <CustomizedDefaultButton
      aria-label="10"
      className="contextual-menu-button hide-mobile effectivemess-measurement-voting-button"
      iconProps={
        {
          "iconName": "CircleRing",
        }
      }
      onClick={[Function]}
    />
  </td>
</tr>
`;
